### PR TITLE
added check-internet-speed v0.0.3

### PR DESCRIPTION
**Pull request by sujitmandal**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Review by sujitmandal — COMMENTED** [review]
please merge the commit

**Comment by traversaro** [comment]
Hello @sujitmandal, please change the PR and the description of the PR to mention which package you are adding. Once that is done, please follow https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#12-my-pull-request-passes-all-checks-but-hasnt-received-any-attention-how-do-i-call-attention-to-my-pr--what-is-the-customary-amount-of-time-to-wait to ask for a review.

**Review by sujitmandal — COMMENTED** [review]
checking internet upload and download speed.

**Comment by sujitmandal** [comment]
Hello @traversaro Please revew the updated code.

**Comment by sujitmandal** [comment]
@traversaro updated please

**Comment by traversaro** [comment]
Please check the first point of the checklist, i.e. " Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"." At the moment the title is a generic "added new package", that does not describe what the PR is doing.

**Comment by sujitmandal** [comment]
@traversaro
i have updated the PR name; please check

**Comment by traversaro** [comment]
It seems that this recipe is a `noarch: python`, you can check the instructions in https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python . More in general, I am not a reviewer here, so you may want to follow the steps in https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#12-my-pull-request-passes-all-checks-but-hasnt-received-any-attention-how-do-i-call-attention-to-my-pr--what-is-the-customary-amount-of-time-to-wait for actually getting a review. I suggest to do that after you ensure that your recipe is done as `noarch: python`.

**Comment by sujitmandal** [comment]
@conda-forge/help-python Please review the updated code.

File: recipes/check-internet-speed/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{% set name = "check-internet-speed" %}
```
Comment (ocefpaf) [inline recipes/check-internet-speed/meta.yaml:1]: Please use grayskull to generate Python recipes. Grayskull make it easier for us to review.

See https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#grayskull---recipe-generator-for-python-packages-on-pypi for more information.

Comment (sujitmandal) [reply recipes/check-internet-speed/meta.yaml:1]: @ocefpaf i have regenerated Python recipes useing grayskull, Please review

File: recipes/check-internet-speed/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/check-internet-speed-{{ version }}.tar.gz
+  sha256: 8be5c80f0d1f29e07529479cc40b3f2936196cd21be74d31495a818f981af917
+
+build:
+  skip: False  # [win]
+  noarch: python
```
[the remark this answers is not in the document]
Comment (sujitmandal) [reply recipes/check-internet-speed/meta.yaml]: i have added now Windows also

File: recipes/check-internet-speed/meta.yaml [hunk 3]
```diff
@@ -10,18 +10,16 @@ source:
   sha256: 8be5c80f0d1f29e07529479cc40b3f2936196cd21be74d31495a818f981af917
 
 build:
-  #skip: False  # [win]
-  #noarch: python
```
Comment (ocefpaf) [inline recipes/check-internet-speed/meta.yaml]: Please don't remove the noarch.

Comment (sujitmandal) [reply recipes/check-internet-speed/meta.yaml]: if i added noarc, i got this error...


❌ Non noarch packages should have python requirement without any version constraints.
❌ Non noarch packages should have python requirement without any version constraints.

File: recipes/check-internet-speed/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  sha256: 8be5c80f0d1f29e07529479cc40b3f2936196cd21be74d31495a818f981af917
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - setuptools
+    - python
```
Comment (ocefpaf) [inline recipes/check-internet-speed/meta.yaml:19]:
```suggestion
    - python  {{ python_min }}
```

File: recipes/check-internet-speed/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - setuptools
+    - python 
+    - pip
+  run:
+    - python
```
Comment (ocefpaf) [inline recipes/check-internet-speed/meta.yaml:22]:
```suggestion
    - python >={{ python_min }}
```

File: recipes/check-internet-speed/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  run:
+    - python 
+
+test:
+  imports:
+    - check_internet_speed
+  commands:
+    - pip check
+  requires:
+    - pip
```
Comment (ocefpaf) [inline recipes/check-internet-speed/meta.yaml:30]:
```suggestion
    - pip
    - python {{ python_min }}
```

**Comment by ocefpaf** [comment]
Please see https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-builds

**Comment by sujitmandal** [comment]
@ocefpaf I have updated the code, please review

**Comment by ocefpaf** [comment]
> @ocefpaf I have updated the code, please review @conda-forge/help-python

GitHub keeps us updated of your changes. No need to keep pinging the team or me all the time. We are all volunteers doing this in out spare time. Please be patient.

**Pull request merged**